pygeos/tests/test_io.py

```diff
@@ -375,17 +375,24 @@ def test_to_wkb_srid():
 
 
 @pytest.mark.skipif(
-    pygeos.geos_version >= (3, 8, 0), reason="Pre GEOS 3.8.0 has 3D empty points"
+    pygeos.geos_version < (3, 9, 0),
+    reason="Pre GEOS 3.9 empty point serialization is buggy",
 )
 @pytest.mark.parametrize(
     "geom,dims,expected",
     [
         (empty_point, 2, POINT_NAN_WKB),
-        (empty_point, 3, POINTZ_NAN_WKB),
+        (empty_point, 3, POINT_NAN_WKB),
```

Comment:
This behaviour changed? (which seems a correct change though)

Reply:
On GEOS < 3.8, to_wkb(empty_point, output_dimension=3) indeed gave the (faulty) 3D WKB string
On GEOS ==3.8, this works correctly, but you always get a 2D WKB (also for empty_point_z
On GEOS >= 3.9, everything works fine.

I split the test into 3 to make this more clear and updated the docs.